### PR TITLE
handle list input for scenario

### PR DESCRIPTION
**Pull request by nicolejkeeney**
# Description of PR

**See header below "How to test" for instructions on how to test this PR**

 When I wrote the new ```get_data``` function for the climatekitae_direct_data_download notebook (the functions that let you access/download data without the GUI), I wrote it such that you could only retrieve data for a single scenario (because it was way more simple programmatically). 

In working on the warming levels code, I realized we really need to be able to pull multiple scenarios, like the selections GUI does. So, in this update, I allow you to use a list as an input for the scenario object. 

## How to test

In the climakitae_direct_data_download notebook, try retrieving and viewing different data options using the following two functions: ```get_data_options``` and ```get_data```.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

**Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.**

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by nicolejkeeney** [comment]
[Add option to retrieve multiple scenarios in get_data function](https://trello.com/c/wq8bx0Bi/701-add-option-to-retrieve-multiple-scenarios-in-getdata-function)

**Comment by vicford** [comment]
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [2], line 1
----> 1 get_data_options()

File ~/src/climakitae/climakitae/core/data_interface.py:1500, in get_data_options(variable, downscaling_method, resolution, timescale, scenario, tidy)
   1487         return None
   1488 d = {
   1489     "variable": variable if type(variable) == list else [variable],
   1490     "timescale": timescale if type(timescale) == list else [timescale],
   (...)
   1497     "resolution": resolution if type(resolution) == list else [resolution],
   1498 }
-> 1500 d = _check_if_good_input(d, cat_df)
   1502 # Subset the catalog with the user's inputs
   1503 cat_subset = cat_df[
   1504     (cat_df["variable"].isin(d["variable"]))
   1505     & (cat_df["downscaling_method"].isin(d["downscaling_method"]))
   (...)
   1508     & (cat_df["scenario"].isin(d["scenario"]))
   1509 ].reset_index(drop=True)

File ~/src/climakitae/climakitae/core/data_interface.py:1416, in _check_if_good_input(d, cat_df)
   1412         pass
   1414 if val_i not in valid_options:
-> 1416     print("Input " + key + "='" + val_i + "' is not a valid option.")
   1418     closest_options = _get_closest_options(val_i, valid_options)
   1420     # Sad! No closest options found. Just set the key to all valid options

TypeError: can only concatenate str (not "NoneType") to str
```
Error when running `get_data_options()`

**Comment by nicolejkeeney** [comment]
> ```
> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> Cell In [2], line 1
> ----> 1 get_data_options()
> 
> File ~/src/climakitae/climakitae/core/data_interface.py:1500, in get_data_options(variable, downscaling_method, resolution, timescale, scenario, tidy)
>    1487         return None
>    1488 d = {
>    1489     "variable": variable if type(variable) == list else [variable],
>    1490     "timescale": timescale if type(timescale) == list else [timescale],
>    (...)
>    1497     "resolution": resolution if type(resolution) == list else [resolution],
>    1498 }
> -> 1500 d = _check_if_good_input(d, cat_df)
>    1502 # Subset the catalog with the user's inputs
>    1503 cat_subset = cat_df[
>    1504     (cat_df["variable"].isin(d["variable"]))
>    1505     & (cat_df["downscaling_method"].isin(d["downscaling_method"]))
>    (...)
>    1508     & (cat_df["scenario"].isin(d["scenario"]))
>    1509 ].reset_index(drop=True)
> 
> File ~/src/climakitae/climakitae/core/data_interface.py:1416, in _check_if_good_input(d, cat_df)
>    1412         pass
>    1414 if val_i not in valid_options:
> -> 1416     print("Input " + key + "='" + val_i + "' is not a valid option.")
>    1418     closest_options = _get_closest_options(val_i, valid_options)
>    1420     # Sad! No closest options found. Just set the key to all valid options
> 
> TypeError: can only concatenate str (not "NoneType") to str
> ```
> 
> Error when running `get_data_options()`

Good find!! I will investigate!! Do you remember your inputs by chance? @vicford

**Comment by nicolejkeeney** [comment]
> > Good find!! I will investigate!! Do you remember your inputs by chance? @vicford
> 
> I didn't even get to the input stage :( it crapped out just on the call as is!

Lol that's embarassing.... oops.

**Review by vicford — APPROVED** [review]
This is cool @nicolejkeeney  -- was able to test on a few options here
```
get_data(
    variable = "Air Temperature at 2m", 
    downscaling_method = "Dynamical", 
    resolution = "9 km", 
    timescale = "monthly", 
    # time_slice = (1985, 2020),
    scenario = ["SSP 3-7.0 -- Business as Usual", "SSP 2-4.5 -- Middle of the Road"],
    cached_area = "San Bernardino County", 
    area_average = "Yes"
)
```

**and the real bonus here**
```
get_data(
    variable = "Air Temperature at 2m", 
    downscaling_method = "Dynamical", 
    resolution = "9 km", 
    timescale = "monthly", 
    # time_slice = (1985, 2020),
    scenario = ["SSP 3-7.0 -- Business as Usual", "SSP 2-4.5 -- "],
    cached_area = "San Bernardino County", 
    area_average = "Yes"
)
```
which then identified that I wanted "SSP 2-4.5 -- Middle of the Road" which is mondo cool. 

Some general feedback, for future PRs
- We should include an example of multiple scenario retrieval in this notebook to make it explicitly clear
- the code can identify which scenario we're looking for *if there is a space between SSP and the first number*. If someone entered "SSP3-7.0" instead of "SSP 3-7.0" it does throw an error and valid options. But user friendly addition would to ensure "SSP3..." also can guess for "SSP 3-7.0" etc. 
- There is no time slice option in get_data or get_data_options

File: climakitae/core/data_interface.py [hunk 1]
```diff
@@ -1397,76 +1479,35 @@ def get_data_options(
...
-                            + good_resolution_input
-                            + "'"
-                        )
-                        d[key] = [good_resolution_input]
-                        continue
-                except:
-                    pass
-
-            print("Input " + key + "='" + val + "' is not a valid option.")
+    # For the non-None inputs, check if the inputs are good
```
Comment (elehmer) [inline climakitae/core/data_interface.py:1502]: Does this commented out code need to be in?

**Review by elehmer — APPROVED** [review]
Just the one comment. Otherwise looks good.

File: climakitae/core/data_interface.py (hunk 1 above)
Comment (nicolejkeeney) [reply climakitae/core/data_interface.py:1502]: Nope; I will delete, thanks!

**Pull request merged**